src/Decoder.cpp

```diff
@@ -150,6 +153,81 @@ namespace nbs {
         return jsRange;
     }
 
+    Napi::Value Decoder::NextTimestamp(const Napi::CallbackInfo& info) {
+        Napi::Env env = info.Env();
+
+        uint64_t timestamp = 0;
+        try {
+            timestamp = this->TimestampFromJsValue(info[0], env);
+        }
+        catch (const std::exception& ex) {
+            Napi::TypeError::New(env, std::string("invalid type for argument `timestamp`: ") + ex.what())
+                .ThrowAsJavaScriptException();
+            return env.Undefined();
+        }
+
+        // Array of typeSubtypes
+        std::vector<TypeSubtype> types;
+        if (info[1].IsArray()) {
+            auto argTypes = info[1].As<Napi::Array>();
+
+            // Get the types and subtypes
+            for (std::size_t i = 0; i < argTypes.Length(); i++) {
+                auto item = argTypes.Get(i);
+
+                try {
+                    auto typeSubtype = this->TypeSubtypeFromJsValue(item, env);
+                    types.push_back(typeSubtype);
+                }
+
+                catch (const std::exception& ex) {
+                    Napi::TypeError::New(env, "invalid item type in `types` array: " + std::string(ex.what()))
+                        .ThrowAsJavaScriptException();
+                    return env.Undefined();
+                }
+            }
+        }
+        else if (info[1].IsUndefined()) {
+            types = this->index.getTypes();
+        }
+
+        // Single type
+        else {
+            try {
+                types.push_back(this->TypeSubtypeFromJsValue(info[1], env));
+            }
+            catch (const std::exception& ex) {
+                Napi::TypeError::New(
+                    env,
+                    std::string("invalid type for argument `typeSubtype`: invalid `.subtype`: expected number")
+                        + ex.what())
+                    .ThrowAsJavaScriptException();
+                return env.Undefined();
+            }
+        }
+
+        int steps;
+        try {
+            // If steps is undefined default = 1.
+            if (std::isnan(steps)) {
```

Comment:
Move the check for nan after reading the steps argument.

Reply:
Changed.